File: Apps/Codaxy.Dextop.Showcase/Demos/Remoting/Enums.js
Ext.define('Showcase.demos.EnumDemoWindow', {
	extend: 'Dextop.Window',
	width: 300,
	autoHeight: true,
	title: 'Enum',

	initComponent: function () {
		Ext.apply(this, {
			items: [{
				layout: 'hbox',
				//bodyStyle: 'padding: 10px',
				autoHeight: true,
				border: false,
				defaults: {
					margins: '10 5 10 5'
				},
				items: [{
					xtype: 'button',
					text: 'Toggle',
					scope: this,
					handler: function () {
						this.remote.ToggleStatus({
							scope: this,
							type: 'alert',
							success: this.updateStatus 
						});
					}
				}, {
					xtype: 'button',
					text: 'Make Valid',
					scope: this,
					handler: function () {
						this.remote.SetStatus(Showcase.demos.EnumDemoStatus.Valid, {
							scope: this,
							type: 'alert',
							success: this.updateStatus
						});
					}
				}, {
					xtype: 'button',
					text: 'Make Invalid',
					scope: this,
					handler: function () {
						this.remote.SetStatus(Showcase.demos.EnumDemoStatus.Invalid, {
							scope: this,
							type: 'alert',
							success: this.updateStatus
						});
					}
				}, {
					xtype: 'label'
				}]
			}]
		});
		this.callParent(arguments);
	},

	updateStatus: function (status) {
		switch (status) {
			case Showcase.demos.EnumDemoStatus.Valid:
				this.down('label').setText('Valid');
				break;
			case Showcase.demos.EnumDemoStatus.Invalid:
				this.down('label').setText('Invalid');
				break;
		}
	}
});

File: Apps/Codaxy.Dextop.Showcase/Demos/Remoting/Enums.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using Codaxy.Dextop.Remoting;

namespace Codaxy.Dextop.Showcase.Demos.Remoting
{
	[DextopEnum]
	enum EnumDemoStatus { Valid, Invalid }

	[Demo("Enums",
        Title = "Enums",
        Description = "C# enumerations can be mapped to the JS.",
        Path = "~/Demos/Remoting"
    )]
    [LevelMedium]
    [TopicDextopRemoting]
    [CategoryFeature]
    public class EnumDemoWindow : DextopWindow
    {  
		EnumDemoStatus status = EnumDemoStatus.Invalid;

		[DextopRemotable]
		EnumDemoStatus ToggleStatus()
		{
			if (status == EnumDemoStatus.Valid)
				status = EnumDemoStatus.Invalid;
			else
				status = EnumDemoStatus.Valid;
			return status;
		}

		[DextopRemotable]
		EnumDemoStatus SetStatus(EnumDemoStatus v)
		{
			return status = v; 
		}
    }
}